src/utils/utils_pipeline.c

```diff
@@ -39,6 +39,12 @@ static int _ocf_pipeline_run_step(struct ocf_request *req)
 			pipeline->next_step++;
 			step->hndl(pipeline, pipeline->priv, &step->arg);
 			return 0;
+		case ocf_pipeline_step_conditional:
+			arg = &step->args[pipeline->next_arg++];
+			if (step->cond(pipeline, pipeline->priv, arg)) {
+				step->hndl(pipeline, pipeline->priv, arg);	
+				return 0;
+			}			
```

Comment:
We need to increment `pipeline->next_step++`, otherwise in the next iterations we will try to execute the same step over and over again.